Engineering resprite: Part 1 - Suits

SomeAngryMiner opened the pull request:
# What is this Pull Request about?

What it says on the tin.
Its the first part of a 12 part sequential pull request series, starting with the simpler set: suits of all kinds: from radsuits, to vodsuits, and the CE hardsuit.

## what was changed, now?

Radsuit:
Completely redesign. The old one was a vague, cleaned up scrunched up JPEG as far as our information goes, and its dated to times up to 2010(!), depending on the source asked. The new one is based on modern-era CBRN suits, with a few details and artistic liberties to fit on a more sci-fi setting, such as seams and piping, for a more "space-like" look, as it was described.

Anomaly suit:
Complete redesign. While not downstraight terrible, the old anomaly suit suffered from needless noise and vague shapes: most of the texture and shape of the suit was essentially lost the more inwards your eye went. The new one tries to keep a puffy, cohesive shape, along with a more visually different helmet, to counter-balance the overly orange balance of the sprite.

Excavation suit:
Minor edits. This was the main sketch behind the excavation suit on the first resprite, but was replaced with the placeholder that currently exists once we hit the deadline and this one remained unfinished. Proportions have been fixed, and uses a more sensible yellow color. 

Chief Engineer hardsuit:

OH boy. Here we go. this'll probably have huge backlash.

Complete overhaul. The design has been changed to a more distinct shape, keeping the iconicity of the old white suit with a new spin: instead of a ribbed visor, it was replaced with a geometrical dome, along with a more contrasting underfabric, and added engineering high-visibility lines around the joints. Finally, it had its comically thin RIG backpack replaced with a proper life support backpack, to further accentuate a bulky, advanced shape.

## Pictures!
>3.6 IUs. not good, not terrible.
![image](https://user-images.githubusercontent.com/53237389/155256852-b1ca819c-5cef-4c69-90c0-4bad6f0a8595.png)

>Hmmm. I wonder what kind of rock the purples got today.
![image](https://user-images.githubusercontent.com/53237389/155256929-698dbe2b-4848-4afb-8beb-a20e9c126123.png)

>A suit fit for a free man.
![image](https://user-images.githubusercontent.com/53237389/155256964-a8efcf9c-3253-49e0-be91-5c2b6f2680a8.png)

>I am getting my face eaten off so bad for this.
![image](https://user-images.githubusercontent.com/53237389/155257011-cbb0ba34-7406-454e-a649-247a7ab931bd.png)


As usual, ready for conflicts and light overlay weirdness. God bless baystation12.

:cl: AzlanOnPC
add: The CE Hardsuit, Radsuit, Anomaly suit and Excavation suit have received a sprite update.
/:cl:

SomeAngryMiner committed:
Adds new engineering suit sprites

Merge conflict fix part dos

Merge conflict fix

fix batch 1

adds new suit sprites

kill kill bite maim bite kill

Spookerton reviewed (dismissed):
Humans can wear skrell suits; skrell ~~anomaly~~ excavation suit head is missing human configuration -
![https://i.imgur.com/ARZulYT.png](https://i.imgur.com/ARZulYT.png)

Spookerton commented:
And some stuff I'm not super interested in being fixed here, but worth noting for completeness:

Radiation suit helmet has no inhands (inherited) -
![https://i.imgur.com/lxvOQXs.png](https://i.imgur.com/lxvOQXs.png)

Anomaly suit is taller L/R than U/D -
![https://i.imgur.com/I8ufCaR.png](https://i.imgur.com/I8ufCaR.png)

Anomaly suit inhands is the biosuit (inherited) -
![https://i.imgur.com/QYtQf06.png](https://i.imgur.com/QYtQf06.png)

Human excavation suit inhands is a space suit and helmet (inherited) -
![https://i.imgur.com/y4kb6Vu.png](https://i.imgur.com/y4kb6Vu.png)

Refitted excavation suit inhands is missing (inherited) -
![https://i.imgur.com/nFqHw09.png](https://i.imgur.com/nFqHw09.png)

SierraKomodo reviewed (commented):
Some probably subjective feedback on the sprites:

- I and a few other staff members prefer the old excavation suit's style and color choice over the relatively flatter/muted looked new style.
- The new Chief Engineer rig feels a little too busy compared to other suits, but I haven't loaded this up on test to actually compare and see how it looks in-world yet. It might just be me looking at sprites outside of the context of in-game getting to me.
- The helmet on the anomaly suit feels abnormally large and flat.

SomeAngryMiner commented:
>Radiation suit helmet has no inhands (inherited) 

Fixed. Radiation suits and hoods now have proper inhands.

> Anomaly suit is taller L/R than U/D

Fixed. They're both the same height, now.

>  Anomaly suit inhands is the biosuit (inherited) 

Fixed. They have unique inhands, now.

> Human excavation suit inhands is a space suit and helmet (inherited)

Fixed. Unique inhands have been made, too.

> The new Chief Engineer rig feels a little too busy compared to other suits, but I haven't loaded this up on test to actually compare and see how it looks in-world yet. It might just be me looking at sprites outside of the context of in-game getting to me.

Ingame, due to how hardsuit layers work, it looks definitely busier, but still within reasonable boundaries: this is because of how hardsuit layers work: unlike voidsuits, or any other gear, they will hide belts, backpacks, goggles, masks and whatever, making that the one and only thing you'll see, besides light overlays, damage decals and inhands.

> The helmet on the anomaly suit feels abnormally large and flat.

Added transparency and details for the flatness. Abnormal height is intentional: most if not all the helmets so far have been slim and/or form-fitting, and i think breaking that with the chunky suit made to withstand the experiment process with things  of likely unknown effects is a good idea. Can be changed if they so want it.

> I and a few other staff members prefer the old excavation suit's style and color choice over the relatively flatter/muted looked new style.

Reasoning behind this is simple: the other suit was overly done on certain parts, namely chest and arms; That, lacking the standarization of the hardsuit, will reveal stuff like belts, backpacks and masks if worn around the neck, making the sprite in general _too_ busy. 

Shape in general was flawed: As of right now, the current suit has a balance issue of leaning too much into the front, and lacking any back, making it look like its leaning forward/standing on their tip toes. 

Color choice was for the enviroment: this suit will generally be in dull or generally unclean areas (see: asteroids), and having a squeaky, brilliant suit wont fit the general area it will be found on. 
This new "dull" yellow is also used for less sterile/advanced equipment, as seen on the powerloader colors it was sampled on, and its usage on the radsuit too.  

If its wanted, i can add coloring or stripes like the old version.

>Refitted excavation suit inhands is missing (inherited) - Humans can wear skrell suits; skrell anomaly suit head is missing human configuration

Disturbingly, i could not reproduce this locally, and all skrell suits had their proper human helmets/inhands.
As for the humans being able to wear skrell suits, it seems to be an universal issue with cyclers: any suit that passes through a cycler to fit skrell, will also be able to worn by humans without consequence. This is handled with code i'm not familiar with, but from a glance it needs enough cleaning up to need its own PR.

Did you know only the merc suit cycler is the only one meant to repair suits? yeah, that's apparently an old check broken on the same code, too.

Spookerton commented:
> As for the humans being able to wear skrell suits, it seems to be an universal issue with cyclers: any suit that passes through a cycler to fit skrell, will also be able to worn by humans without consequence. This is handled with code i'm not familiar with, but from a glance it needs enough cleaning up to need its own PR.

Humans being able to wear skrell suits is intentional. Not sure why you couldn't reproduce it. I'll take another look.

Spookerton reviewed (dismissed):
I've seen some pushback on the style of at least one suit but the skrell/human oddness is the only part that stands out in technical terms, and should be resolvable entirely with later code.

SierraKomodo reviewed (dismissed):
Merge conflicts

SomeAngryMiner commented:
Conflict with #31748 fixed. After a quick integrity check no changes made by other PRs were pushed out by the rebase.

Spookerton reviewed (dismissed):
re-approve on prior basis

fwiw I think there's a few things that suffer from the same thing and the different species, different sheet thing is crusty and stretched by current behaviors. todo self all that stuff.
